lib/foreman_maintain/utils/service/abstract.rb

```diff
@@ -66,6 +66,14 @@ def disable
       def running?
         raise NotImplementedError
       end
+
+      def instance_parent_unit
```

Comment:
Here, you defined a method `instance_parent_unit` that returns `parent_unit` value from options. Instead of method, I will prefer to use `attr_reader` that ruby provides in the `Systemd` class.
 
Refer similar example from `RemoteDB` class - https://github.com/theforeman/foreman_maintain/blob/master/lib/foreman_maintain/utils/service/remote_db.rb#L4

Reply:
Resolved